src/core/src/op/if.cpp

```diff
@@ -33,15 +33,21 @@ static ov::PartialShape resolve_shape(const ov::PartialShape& then_pshape, const
         return ov::PartialShape::dynamic();
     }
     if (then_rank.get_length() != else_rank.get_length()) {
+        auto isScalar = [](const ov::PartialShape& pshape) {
+            return ((pshape.rank() == 0) || (pshape.rank() == 1 && pshape.get_shape()[0] == 1));
+        };
         // Union of scalar and 1D case
         if (then_rank.get_length() <= 1 && else_rank.get_length() <= 1) {
+            if (isScalar(then_pshape) && isScalar(else_pshape)) {
+                return ov::PartialShape{1};
+            }
             return ov::PartialShape::dynamic(1);
```

Comment:
```suggestion
            return () ? ov::PartialShape{1} : ov::PartialShape::dynamic(1);
```
Can there be test added/updated e.g. in `src/core/tests/type_prop/if.cpp`